### DISCUSSION TITLE
Can ukv be used as object store?

### DISCUSSION THREAD

**Discussion by kesavkolla** [unum-cloud/UStore#257] [Q&A]
Can we keep large binary objects in ukv? The traditional rocksdb has issues with blob data insertions. I remember seeing it somewhere that rocksdb wal will have issues with large values of binary data. Will ukv has any issues with binary objects?

**Comment by ashvardanian** [comment]
UKV inherits the strengths and weaknesses of the underlying Key-Value Store. Luckily, its swappable. Still, if you are using the RocksDB backend, you can disable WAL at the transaction level, essentially avoiding WAL for the heavy object insertions.

**Comment by kesavkolla** [comment]
Does this work well with udisk? I am curiously waiting for udisk implementation to be released so that we can test it out.

**Reply by ashvardanian** [reply]
How large your BLOBs would be?

**Reply by kesavkolla** [reply]
Arbitrary. It can range from few MB to few GB. I have seen some blob data with 10GB. It's my application I can put an artificial upper limit too. I can code if the size is more than upper limit split it into multiple.

**Reply by ashvardanian** [reply]
In UDisk we support entries up to 4GB, but the first version works best with small entries under 100 KB. The version that will arrive in Q3 will be truly universal.
